src/partition/tests/ReceivedPartitionTest.cpp

```diff
@@ -1087,6 +1087,147 @@ BOOST_AUTO_TEST_CASE(TestCompareBoundingBoxes3D)
+      position << -2.0, -1.0;
+      mesh->createVertex(position);
+    } else {
+      Eigen::VectorXd position(dimensions);
+      position << 0.0, 0.0;
+      mesh->createVertex(position);
+      position << 1.0, 0.0;
+      mesh->createVertex(position);
+      position << 2.0, 0.0;
+      mesh->createVertex(position);
+      position << -0.75, -1.0;
+      mesh->createVertex(position);
+      position << 0.0, -1.0;
+      mesh->createVertex(position);
+      position << 1.0, -1.0;
+      mesh->createVertex(position);
+      position << 2.0, -1.0;
+      mesh->createVertex(position);
+    }
+  }
+  mesh->computeState();
+  mesh->computeBoundingBox();
+
+  if (context.isNamed("Solid")) {
+    m2n->createDistributedCommunication(mesh);
+    ProvidedPartition part(mesh);
+    part.addM2N(m2n);
+
+    part.compareBoundingBoxes();
+
+    if (context.isMaster()) {
+      BOOST_TEST(mesh->getConnectedRanks().size() == 2);
+      BOOST_TEST(mesh->getConnectedRanks()[0] == 0);
+      BOOST_TEST(mesh->getConnectedRanks()[1] == 1);
+    } else {
+      BOOST_TEST(mesh->getConnectedRanks().size() == 1);
+      BOOST_TEST(mesh->getConnectedRanks()[0] == 1);      
+    }
+
+    m2n->acceptSlavesPreConnection("FluidSlaves", "SolidSlaves");
+
+    part.communicate();
+    part.compute();
+
+    if (context.isMaster()) {
+      BOOST_TEST(mesh->getCommunicationMap()[0][0] == 0);
+      BOOST_TEST(mesh->getCommunicationMap()[0][1] == 1);
+      BOOST_TEST(mesh->getCommunicationMap()[0][2] == 2);
+
+      // std::cout<<"master sends vertices: " << mesh->getCommunicationMap()[0] << " to rank 0 and verticees " << mesh->getCommunicationMap()[1] << " to rank 1 " << std::endl; 
+      
+      
+    } else {
+      // BOOST_TEST(mesh->getCommunicationMap()[0].size() == 0);
+      BOOST_TEST(mesh->getCommunicationMap()[1][0] == 0);
+      BOOST_TEST(mesh->getCommunicationMap()[1][1] == 1);
+      BOOST_TEST(mesh->getCommunicationMap()[1][2] == 2);
+      // std::cout<<"slave sends vertices: " << mesh->getCommunicationMap()[0] << " to rank 0 and verticees " << mesh->getCommunicationMap()[1] << " to rank 1 " << std::endl; 
+    }
+  } else {
+    m2n->createDistributedCommunication(receivedMesh);
+    mapping::PtrMapping boundingFromMapping = mapping::PtrMapping(new mapping::NearestNeighborMapping(mapping::Mapping::CONSISTENT, dimensions));
+    mapping::PtrMapping boundingToMapping   = mapping::PtrMapping(new mapping::NearestNeighborMapping(mapping::Mapping::CONSERVATIVE, dimensions));
+    boundingFromMapping->setMeshes(receivedMesh, mesh);
+    boundingToMapping->setMeshes(mesh, receivedMesh);
+
+    ReceivedPartition part(receivedMesh, ReceivedPartition::ON_SLAVES, safetyFactor);
+
+    part.addM2N(m2n);
+
+    part.addFromMapping(boundingFromMapping);
+    part.addToMapping(boundingToMapping);
+
+    part.compareBoundingBoxes();
+
+    m2n->requestSlavesPreConnection("FluidSlaves", "SolidSlaves");
+
+    part.communicate();
+    part.compute();
```

Comment:
Shouldn't we test whether the actual owner information is as expected?

Reply:
Right, test is modified.